akka-projection-eventsourced/src/main/scala/akka/projection/eventsourced/javadsl/EventSourcedProvider.scala

```diff
@@ -30,16 +30,16 @@ import akka.persistence.query.typed.javadsl.LoadEventQuery
 import akka.projection.BySlicesSourceProvider
 import akka.projection.eventsourced.EventEnvelope
 import akka.projection.internal.CanTriggerReplay
-import akka.projection.javadsl
-import akka.projection.javadsl.SourceProvider
+import akka.projection.javadsl.CustomStartOffsetSourceProvider
+import akka.projection.javadsl.CustomStartOffsetSourceProviderImpl
 import akka.stream.javadsl.Source
 
 object EventSourcedProvider {
 
   def eventsByTag[Event](
       system: ActorSystem[_],
       readJournalPluginId: String,
-      tag: String): SourceProvider[Offset, EventEnvelope[Event]] = {
+      tag: String): CustomStartOffsetSourceProvider[Offset, EventEnvelope[Event]] = {
```

Comment:
I think it's fine to change the signature like this in a compatible way. mima doesn't complain. Static methods have their own rules though. Might have to try it out.

Alternative would be to add it to SourceProvider as mentioned above.

Another alternative would be to add additional overloaded methods with the `loadFromOffsetStore` and `startOffset` parameters.

Reply:
actually, mima complains about this (when running in right project 🙂 )
```
[error]  * static method eventsBySlices(akka.actor.typed.ActorSystem,java.lang.String,java.lang.String,Int,Int)akka.projection.scaladsl.SourceProvider in class akka.projection.eventsourced.scaladsl.EventSourcedProvider has a different result type in current version, where it is akka.projection.scaladsl.CustomStartOffsetSourceProvider rather than akka.projection.scaladsl.SourceProvider
```